src/details/ArborX_DetailsDistributedTreeImpl.hpp

```diff
@@ -42,6 +42,70 @@ struct DefaultCallbackWithRank
   }
 };
 
+template <class Predicates, class Distances>
+struct WithinDistanceFromPredicates
+{
+  Predicates predicates;
+  Distances distances;
+};
+
+} // namespace Details
+
+template <class Predicates, class Distances>
```

Comment:
Why do you need `Distances`?

Reply:
I think the idea is that you can use an array, or a generator returning the same distance for all.